### PR TITLE
XFRM-214 Make OpenMRS XForms Module support PostgreSQL

### PR DESCRIPTION
**Pull request by Am-Coder**
**Changes I made**
1. Created liquibase file.
2. Changed use of 0 to false for boolean columns
3. Chages storage of tabIndex column as tab_index in db.
4. Mapped sequences correctly.

**Link to ticket**
https://issues.openmrs.org/browse/XFRM-214

### CODE REVIEW

**Comment by Am-Coder** [comment]
Changes have been tested on both PostgreSQL and MySQL and everything is working fine.

File: api/src/main/resources/MedicalHistoryField.hbm.xml [hunk 1]
```diff
@@ -13,7 +13,12 @@
 		
 		<property name="name" type="java.lang.String" length="255" not-null="true"/>
 		
-		<property name="tabIndex" type="int" />
+		<!-- tabIndex property in database is mapped to tab_index.
+		When sqldiff.xml was used it was stored as tabIndex in db. 
+		This caused problem in PostgreSQL because Postgres converts all column 
+		names to lower case when fetching from db and thus no result was returned since
+		column tabIndex was being searched as tabindex and thus was never found -->
+		<property name="tabIndex" column="tab_index" type="int" />
```
Comment (dkayiwa) [inline api/src/main/resources/MedicalHistoryField.hbm.xml:21]: What happens to those who already had the old column name?

Comment (Am-Coder) [reply api/src/main/resources/MedicalHistoryField.hbm.xml]: That will be a problem. Thanks for pointing that out. Let me see if I can find another solution for it.

File: api/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    <changeSet author="aman" id="xforms-25082020">
+    	<comment> Adding uuids to all data in the the xform table </comment>
+		<sql>update xforms_xform set uuid = uuid() where uuid is null;</sql>
+    </changeSet>
+    <changeSet author="aman" id="xforms_update_columnname-31082020">
+        <preConditions onFail="MARK_RAN">
+    		<columnExists tableName="xforms_medical_history_field" columnName="tabIndex"></columnExists>
+   		</preConditions>
+   		<comment> Rename tabIndex to tab_index for table xforms_medical_history_field, Present since 4.3.13 </comment>
+	    <renameColumn columnDataType="INT" newColumnName="tab_index" oldColumnName="tabIndex" tableName="xforms_medical_history_field"/>
```
Comment (Am-Coder) [inline api/src/main/resources/liquibase.xml:235]: @dkayiwa , for instances that already have column "tabIndex" , I have added this changeset to rename the column to "tab_index" . So, now there should not be a problem if one updates to this version of xform.

**Comment by dkayiwa** [comment]
@Am-Coder how did you remember preconditions here but not in the other modules? 😊

**Comment by Am-Coder** [comment]
@dkayiwa , apparently there were a lot of things going in my head at that time. I completely based this PR liquibase structure from Patient Flags liquibase file and thought that I would have done the same for other liquibase files as well. Turns out that was not the case :smile: . Sorry, for this carelessness . Glad Mark pointed that out.

**Comment by dkayiwa** [comment]
It once in a while happens to all of us. 😊

**Pull request merged**